js/modules/modal.js

```diff
@@ -56,5 +67,77 @@ export default function initModal() {
         modalDetail.classList.remove("doNotShow");
       }
     });
+    gettingSupportValue(modalDetail);
+  }
+
+  async function gettingSupportValue(element) {
+    const inputDetails = element.querySelector(".input-pledge");
+    const btnModalSelect = element.querySelector(".btn-modal-select");
+
+    if (errorActive) {
+      const spanDelete = document.querySelector(".error");
+      spanDelete.remove();
+    }
+
+    inputDetails.addEventListener("input", formatCurrency);
+    btnModalSelect.addEventListener("click", () =>
+      gettingPlanId(inputDetails, element)
+    );
+
+    inputDetails.value = "R$";
+
+    function formatCurrency({ target }) {
+      let value = target.value.replace(/\D/g, "");
+      value = (value / 100).toFixed(2).replace(".", ",");
+      value = "R$ " + value;
+      target.value = value;
+    }
+  }
+
+  async function gettingPlanId(inputDetails, element) {
+    const nameInput = inputDetails.name.split("-")[1];
+    try {
+      const valueFormattingInput = inputDetails.value.replace(/\D/g, "");
+      const response = await api.get("/planos");
+      const plans = response.data;
```

Comment:
se sempre vai existir, vc pode passar o conteudo diretamente
```suggestion
      const { data } = await api.get("/planos");
```